azurerm/internal/services/resource/resource_arm_provider_registration.go

```diff
@@ -0,0 +1,188 @@
+package resource
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/profiles/2017-03-09/resources/mgmt/resources"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmResourceProviderRegistration() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmResourceProviderCreateUpdate,
+		Read:   resourceArmResourceProviderRead,
+		Update: resourceArmResourceProviderCreateUpdate,
+		Delete: resourceArmResourceProviderDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"resource_provider_namespace": {
```

Comment:
this wants to be `name` rather than `resource_provider_namespace` - whilst `namespace` is technically correct with regard to this API - every other resource uses `name` - so we may as well make this consistent here:

```suggestion
			"name": {
```